toolchain/check/import_ref.cpp

```diff
@@ -32,16 +32,26 @@ static auto InternalAddImportIR(Context& context, SemIR::ImportIR import_ir)
   return context.import_irs().Add(import_ir);
 }
 
-auto SetApiImportIR(Context& context, SemIR::ImportIR import_ir) -> void {
+// Sets the IR for a specific ImportIRId. Should be called in the right order in
+// order to ensure the correct ID is assigned.
+static auto SetImportIR(Context& context, SemIR::ImportIR import_ir,
+                        SemIR::ImportIRId expected_import_ir_id) -> void {
   auto ir_id = SemIR::ImportIRId::None;
   if (import_ir.sem_ir != nullptr) {
     ir_id = AddImportIR(context, import_ir);
   } else {
     // We don't have a check_ir_id, so add without touching check_ir_map.
     ir_id = InternalAddImportIR(context, import_ir);
   }
-  CARBON_CHECK(ir_id == SemIR::ImportIRId::ApiForImpl,
-               "ApiForImpl must be the first IR");
+  CARBON_CHECK(ir_id == expected_import_ir_id,
+               "Actual ImportIRId ($0) != Expected ImportIRId ({1})", ir_id,
+               expected_import_ir_id);
+}
+
+auto SetApiImportIRs(Context& context, SemIR::ImportIR import_ir) -> void {
```

Comment:
Previously `SetApiImportIR` only set the API file's `ImportIR`, thus the name.

Similarly, `AddImportIR` is significantly different from `SetImportIR`: the former is used with any IR, whereas the latter is only used with the special `ImportIRId` values.

Maybe renaming this to `SetSpecialImportIRs` instead of `SetApiImportIRs` would help reflect that it's now more than API's IR? Similarly, perhaps use the singular form `SetSpecialImportIR` instead of `SetImportIR`?

Reply:
Done.